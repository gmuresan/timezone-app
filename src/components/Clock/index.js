import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';


class Clock extends React.Component {
  static propTypes = {
    gmtOffset: PropTypes.number.isRequired,
  };

  constructor(props) {
    super(props);
    const time = moment();
    time.utcOffset(props.gmtOffset);
    this.state = {
      date: time,
    };
  }

  componentDidMount() {
    this.timerID = setInterval(() => {
      const time = moment();
      time.utcOffset(this.props.gmtOffset);
      this.setState({
        date: time,
      });
    }, 1000);
  }

  componentWillUnmount() {
    clearInterval(this.timerID);
  }

  render() {
    return (
      <div>
        {this.state.date.format('hh:mm:ss A')} GMT
      </div>
    );
  }
}

export default Clock;

