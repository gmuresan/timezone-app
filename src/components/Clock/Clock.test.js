/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';
import Clock from './';

describe('Clock', () => {
  test('renders clock correctly', () => {
    Date.now = jest.fn(() => 1487076708000);
    const wrapper = renderer
      .create(
        <App context={{ insertCss: () => {} }}>
          <Clock gmtOffset={1} />
        </App>,
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});

