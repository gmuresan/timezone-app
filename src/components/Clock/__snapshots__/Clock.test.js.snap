// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clock renders clock correctly 1`] = `
<div>
  01:51:48 PM
   GMT
</div>
`;
